components/tidb_query/src/rpn_expr/impl_math.rs

```diff
@@ -463,6 +463,46 @@ pub fn round_dec(arg: &Option<Decimal>) -> Result<Option<Decimal>> {
     }
 }
 
+#[inline]
+#[rpn_fn]
+pub fn truncate_real_with_int(arg0: &Option<Real>, arg1: &Option<Int>) -> Result<Option<Real>> {
+    match (arg0, arg1) {
+        (&Some(x), &Some(d)) => {
+            let d = if d >= 0 {
+                d.min(i64::from(i32::max_value())) as i32
```

Comment:
Could you please explain why we need to compare `d` with the bounds of `i32`? In `TiDB`, it just converts `i64` into `i32` with overflow.

Reply:
I referred the implementation of non-vectorized truncate_real in TiKV.
And I found the [discuss](https://github.com/tikv/tikv/pull/3532#issuecomment-418701722) in PR #3532 about `i64` to `i32` with overflow.


For example, the result of `truncate(123.1234, 0x10ffffff_00000001)` should be 123.1234.
However, because `int32(0x10ffffff_00000001)` is 1, the result will be 123.1.